flatpak_builder_lint/checks/metainfo.py

```diff
@@ -67,6 +67,12 @@ def _validate(self, path: str, appid: str) -> None:
         ):
             if not os.path.exists(appinfo_icon_path):
                 self.errors.add("appstream-missing-icon-file")
+            if not appstream.has_icon_key(appstream_path):
+                self.errors.add("appstream-missing-icon-key")
```

Comment:
Just to make sure, we're running this only on desktop and console apps?

Reply:
Only for desktop and desktop-application types in appstream. Some console apps ship with no icon key in metainfo or install an icon.